questionable/indexing.nim

```diff
@@ -1,16 +1,23 @@
 import std/macros
 
-macro `.?`*(expression: seq | string, brackets: untyped{nkBracket}): untyped =
+macro `.?`*(expression: seq | string | openArray, brackets: untyped{nkBracket}): untyped =
   # chain is of shape: (seq or string).?[index]
   let index = brackets[0]
   quote do:
     block:
       type T = typeof(`expression`[`index`])
-      let evaluated = `expression`
-      if `index` < evaluated.len:
-        evaluated[`index`].some
+
+      when typeof(`expression`) is openArray:
+        if `index` >= `expression`.low and `index` <= `expression`.high:
```

Comment:
You're using `expression` a number of times here. Because this is a macro, that means that if you do something like this: ``slowComputationThatReturnsOpenArray().?[0]``, then it will evaluate the slow computation each and every time. That is why there a ```let evaluated = `expression` ``` below :smile:

Reply:
Good point, I think I found a way to get around that. I added this commit to the PR, hope you don't mind: 48e8fe8

Reply:
Oops, that didn't work for Nim 1.2.x. Updated in a884e5b.

Reply:
Awesome, thanks for the change 🙏